src/log/ModDaemon.cc

```diff
@@ -298,29 +298,34 @@ logfile_mod_daemon_writeline(Logfile * lf, const char *buf, size_t len)
         }
         return;
     }
-    /* Append this data to the end buffer; create a new one if needed */
+
     /* Are we eol? If so, prefix with our logfile command byte */
+    if (ll->eol) {
+	char tb[2];
+	tb[0] = 'L';
+	tb[1] = '\0';
+	logfile_mod_daemon_append(lf, tb, 1);
+	ll->eol = 0;
+    }
+
+    /* Append this data to the end buffer; create a new one if needed */
     logfile_mod_daemon_append(lf, buf, len);
 }
 
 static void
 logfile_mod_daemon_linestart(Logfile * lf)
 {
-    l_daemon_t *ll = static_cast<l_daemon_t *>(lf->data);
-    char tb[2];
-    assert(ll->eol == 1);
-    ll->eol = 0;
-    tb[0] = 'L';
-    tb[1] = '\0';
-    logfile_mod_daemon_append(lf, tb, 1);
+    /* 
+     * Sending the starting command byte has been moved to
+     * logfile_mod_daemon_writeline
+     */
```

Comment:
Let's document the resulting code rather than the changes that lead to it:

```suggestion
    // logfile_mod_daemon_writeline() sends the starting command
```

This will also fix a CI [failure](https://semaphoreci.com/squid/squid/branches/pull-request-576/builds/2).

----

P.S. You may receive two copies of this change request. AFAICT, GitHub ate my first version, so I am resubmitting it.